src/Illuminate/Support/Collection.php

```diff
@@ -1089,6 +1089,10 @@ public function random($amount = null)
             return Arr::random($this->items);
         }
 
+        if ($amount === 0) {
```

Comment:
If there are 0 items in the collection, we'll never reach this check even though that would seem to be valid with this new logic.